.github/workflows/ci-aws.yml

```diff
@@ -0,0 +1,259 @@
+
+  docker-build:
+    name: docker-build
+    needs: [changes]
+    if: success() #&& (github.event_name == 'push')
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
+        service: ${{ fromJSON(needs.changes.outputs.services) }} 
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+            
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }} 
+          aws-region: us-east-1
+          
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set ecr repository
+        run: |
+          sh .github/workflows/scripts/get_ecr_repository.sh
+        env:
+          SERVICE_NAME: ${{ matrix.service }}
+      - name: Set hash
+        env:
+          working-directory: packages/${{ matrix.service }}
+        run: |
+          GITHUB_HASH_MOD=$(find ./packages/${{ matrix.service }} -type f \( -exec sha1sum {} \; \) | awk '{print $1}' | sort | sha1sum)
+          echo "GITHUB_HASH=${GITHUB_HASH_MOD%???}" >> $GITHUB_ENV
+          echo $GITHUB_HASH
+      - name: Check if repo exist
+        working-directory: packages/${{ matrix.service }}
+        
+        run: |
+          aws ecr describe-repositories --repository-names $ECR_REPOSITORY || aws ecr create-repository --repository-name $ECR_REPOSITORY
+      - name: Check if tag already exist
+        id: tag-exist
+        run: |
+          set +e
+          cmd="$(aws ecr describe-images --repository-name=$ECR_REPOSITORY --image-ids=imageTag=$GITHUB_HASH)"
+          if [[ ! -z "$cmd" ]]
+          then  
+            echo "tag already exist"
+            echo "::set-output name=tag_exist::true"
+          else
+            echo "tag doesn't exist"
+            echo "::set-output name=tag_exist::false"
+          fi
+      
+      - name: set branch-name
+        run: |
+          if [[ $GITHUB_REF != 'refs/heads/master' ]]
+          then 
+            echo "BRANCH_NAME=$(echo $GITHUB_REF_NAME | sed "s/\//-/g")" >> $GITHUB_ENV
+          fi
+
+      - name: Build, tag, and push image to Amazon ECR
```

Comment:
there is GH action that performs that step why not using it (it will be more readable and maintained)

Reply:
I saw it, not sure how it works, if it's not very critical I would like to use our code where I can control exactly what happens.